.github/workflows/main.yaml

```
@@ -0,0 +1,296 @@
+            SOURCE_IMAGE_NAME: "fedora/postgresql-13"
+            SOURCE_IMAGE_TAG: "13"
+          - service: aries-endorser-backup
+            GIT_REPO_URL: BCDevOps/backup-container
+            GIT_REF: 2.5.1
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: docker    # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: artifacts.developer.gov.bc.ca/docker-remote/
+            SOURCE_IMAGE_NAME: centos/postgresql-13-centos7
+            SOURCE_IMAGE_TAG: 20210722-70dc4d3
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+          - service: aries-endorser-proxy
+            GIT_REF: ""
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: proxy     # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: "artifacts.developer.gov.bc.ca/docker-remote/"
+            SOURCE_IMAGE_NAME: caddy
+            SOURCE_IMAGE_TAG: latest
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+          - service: aries-endorser-api
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            GIT_REF: ""
+            DOCKER_FILE_PATH: Dockerfile.endorser # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: endorser # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: artifacts.developer.gov.bc.ca/docker-remote/
+            SOURCE_IMAGE_NAME: python
+            SOURCE_IMAGE_TAG: 3.10-slim-buster
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+    outputs:
+      aries-endorser-agent_digest: ${{ steps.digest.outputs.aries-endorser-agent_digest }} 
+      aries-endorser-backup_digest: ${{ steps.digest.outputs.aries-endorser-backup_digest }} 
+      aries-endorser-api_digest: ${{ steps.digest.outputs.aries-endorser-api_digest }}
+      aries-endorser-proxy_digest: ${{ steps.digest.outputs.aries-endorser-proxy_digest }}
+      aries-endorser-db_digest: ${{ steps.digests.outputs.aries-endorser-db_digest }}  
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with: 
+        repository: ${{ matrix.GIT_REPO_URL }}
+        ref: ${{ matrix.GIT_REF }}
+        
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to image registry
+      if: matrix.REGISTRY_USERNAME_SECRET_NAME  != ''
```

Recommendation to resolve the issue you ran into with this:

```suggestion
      if: matrix.REGISTRY_USERNAME_SECRET_NAME  != '' && matrix.SOURCE_IMAGE_REGISTRY != ''
```